### PR TITLE
Make CDI clone strategy strings correct types

### PR DESCRIPTION
**Pull request by awels**
Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:
Noticed the consts were strings instead of the right types this changes the type to the right type and modifies the usage to not have to cast to the right type all over the place.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
BugFix: Modified CDICloneStrategy strings to be correct type
```

### CODE REVIEW

File: staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/types.go [hunk 1]
```diff
@@ -654,13 +654,13 @@ type CDICloneStrategy string
 
 const (
 	// CloneStrategyHostAssisted specifies slower, host-assisted copy
-	CloneStrategyHostAssisted = "copy"
+	CloneStrategyHostAssisted CDICloneStrategy = "copy"
```
Comment (brybacki) [inline staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/types.go:657]: In yaml it is still a string so no problem that we are changing it here, right?

Comment (awels) [reply staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/types.go:657]: Right it is actually still a 'string' in memory, it is just properly types as something besides a string in the code. This way we don't have to do as many casts to the data structure type.

**Pull request merged**